test-framework/test-suites/unit/tests/switch/pylib/test_switch_pylib_m7800.py

```diff
@@ -49,63 +67,308 @@ def test_install_firmware(self, mock_expectmore):
+		"""Expect an error to be raised due to a missing success indicator and the error message to be captured."""
+		test_switch = SwitchMellanoxM7800('fakeip')
+		firmware_url = f'{test_switch.SUPPORTED_IMAGE_FETCH_PROTOCOLS[0]}://sometrustworthysite.ru'
+
+		error_message = '% unauthorized'
+		mock_expectmore.return_value.ask.return_value = ['other junk', error_message,]
+		with pytest.raises(SwitchException) as exception:
+			test_switch.image_fetch(url = firmware_url)
+		assert(error_message in str(exception))
+
+	def test_image_fetch_error_without_message(self, mock_expectmore):
+		"""Expect an error to be raised due to a missing success indicator.
+
+		The error message is not captured because it is missing the beggining % symbol.
+		"""
+		test_switch = SwitchMellanoxM7800('fakeip')
+		firmware_url = f'{test_switch.SUPPORTED_IMAGE_FETCH_PROTOCOLS[0]}://sometrustworthysite.ru'
+
+		irrelevant_output = ['other junk', 'irrelevant',]
+		mock_expectmore.return_value.ask.return_value = irrelevant_output
+		with pytest.raises(SwitchException) as exception:
+			test_switch.image_fetch(url = firmware_url)
+		assert(not any((output in str(exception) for output in irrelevant_output)))
+
 	def test_show_images(self, mock_expectmore):
 		"""Expect this to try to list the current and available firmware images."""
 		test_switch = SwitchMellanoxM7800('fakeip')
-		test_switch.show_images()
+		with open('/export/test-files/switch/m7800_show_images_console.txt') as test_file:
+			test_console_response = test_file.read().splitlines()
+
+		expected_data = (
+			{
+				1: 'X86_64 3.6.4006 2017-07-03 16:17:39 x86_64',
+				2: 'X86_64 3.6.4006 2017-07-03 16:17:39 x86_64',
+			},
+			2,
+			1,
+			[
+				('image-X86_64-3.6.2002.img', 'X86_64 3.6.2002 2016-09-28 21:00:15 x86_64'),
+				('image-X86_64-3.6.3004.img', 'X86_64 3.6.3004 2017-02-05 17:31:53 x86_64'),
+				('image-X86_64-3.6.4006.img', 'X86_64 3.6.4006 2017-07-03 16:17:39 x86_64'),
+				('image-X86_64-3.6.5009.img', 'X86_64 3.6.5009 2018-01-02 07:42:21 x86_64'),
+				('image-X86_64-3.6.8010.img', 'X86_64 3.6.8010 2018-08-20 18:04:19 x86_64'),
+			],
+		)
+		mock_expectmore.return_value.ask.return_value = test_console_response
+		assert(test_switch.show_images() == expected_data)
 		mock_expectmore.return_value.ask.assert_called_with('show images')
 
 	def test_show_images_no_data(self, mock_expectmore):
-		"""Test that default data is returned when nothing is returned from the switch."""
-		expected_data = {
-			'installed_images': [],
-			'last_boot_partition': None,
-			'next_boot_partition': None,
-			'images_fetched_and_available': [],
-		}
+		"""Test that an error is raised when nothing is returned from the switch."""
 		mock_expectmore.return_value.ask.return_value = []
 		test_switch = SwitchMellanoxM7800('fakeip')
-		assert(test_switch.show_images() == expected_data)
+		with pytest.raises(SwitchException):
+			test_switch.show_images()
+
+	def test__get_errors(self, mock_expectmore):
```

Comment:
At several places you have `test__bah` with two underscored between test and the rest. I would recommend only one, because that is how everything else is.